node_modules/helpers/modules/memdb/index.js

```diff
@@ -29,10 +30,44 @@ function read(filename) {
     }
 }
 
+function findInHMap(input, storageName) {
```

Comment:
I like your approach and this is a very strong starting point.

But can you elaborate on how you designed `addInHMap()` to work in the overall caching architecture?
Maybe I'm mistaken, but to me it seems `addInHMap()` creates a new memdb entry with some `data` that you pass from somewhere; if the data we are passing to the function isn't cached why are we directly creating a hash of it; and if it is already cached (which is the preferred approach) wouldn't this function just clone said data in a new object instead of creating a reference to it in the cache?

Lets discuss this and see how we can move forward.

Reply:
Let's first discuss the elephant in the room. The [moduling of the hashing function](https://github.com/virlancristian/shaorma-bot/pull/4/commits/234722767995aafbc740fe1128a7ed009f3a8b93#r1348163641) and the [refactoring of the hash map modules](https://github.com/virlancristian/shaorma-bot/pull/4/commits/234722767995aafbc740fe1128a7ed009f3a8b93#r1348170535) will be solved after we discuss about the ```addInHMap()``` function.

About the issue of the already cached data, the function doesn't clone the data, since there is a ```for loop``` which iterates through the whole list at the hash key index of the inputed data, and if it finds said data, it will throw an collision error. The only occasion when the data passed through the argument can be cloned is if an user calls the function with a non-corresponding string input type which can give a different hash key other than the one where the data is stored. If you want to prevent this maybe we can work around to create an input sanitizer for the hashing function.
And about the addition of the new data directly into the cache through creating a hash of it, as said in the *convention*, I think it is better for the data in memdb that works with the hash map to be directly cached and stored into their respective files as a hash map instead of a list, so when the program starts the program doesn't waste time on initializing the data lists into the cache and them creating a hash map for them. If you think we should proceed with the latter, or you have better ideas, tell me why and let me know :D